portfolio/src/main/java/com/google/sps/servlets/DailyActivitiesServlet.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+import java.io.IOException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import java.io.FileReader;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+
+/** Returns daily activities data as a JSON object */
+@WebServlet("/daily-activities")
+public class DailyActivitiesServlet extends HttpServlet {
+  // Private variables used to construct json. Represents hours spend on each activity
+  HashMap<String, Double> activityHours;
+
+  @Override
+  public void init() {
+      activityHours = new HashMap<String, Double>();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JSONParser parser = new JSONParser();
+    try {
+        // Parse json file into array that can be worked on
+        Object obj = parser.parse(new FileReader("resources/daily-activities.json"));
+        JSONObject jsonObject = (JSONObject) obj;
+        JSONArray activities = (JSONArray) jsonObject.get("Daily Activities");
+
+        for (int i = 0; i < activities.size(); i++) {
+            JSONObject jsonObj = (JSONObject) activities.get(i);
+
+            // JSONObject inherits keySet() from the Map class. It is used here to
+            // get the keys for each element within the json object (ex. "eating", "reading")
+            jsonObj.keySet().forEach(keyStr ->
+            {
+                // Cannot directly cast Object to double as double is primitive, so long
+                // used as an intermediary
+                updateValues((String) keyStr, Double.valueOf((Long) jsonObj.get(keyStr)));
```

Comment:
can you use [Double](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html) (the class)?

If not, why not use Longs in the first place?  I'm assuming this is for tracking partial hours?  Maybe another option would be to store and track milli/micro secs instead?  This way you wouldn't have fractions.  It'll also be consistent with timestamps and unix time, etc..

Reply:
I've changed it to (Double) jsonObj.get(keyStr). The cause of the of the complication was that I didn't write the hours as decimal numbers in the json file, so jsonObj.get() interpreted it as an integer number. It wouldn't let me directly convert that long to a double, so the Double.valueOf... part on line 56 was like a work-around. 

Fractions are actually what I want, as I want the graph on my page to display the average of all the hours. I feel like using milliseconds might overly complicate it.

Reply:
I see what you mean.  Out of curiosity would `Double.valueOf(jsonObj.get(keyStr))` work?

Both `(Double) something` and `Double.valueOf(something)` does the same thing (converts something to a Double).  I personally prefer the second option since it's more explicit and I feel is clearer for the reader.  That second option might also have safety checks to prevent invalid conversions (C++ does at least, not 100% sure for Java).